matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -500,7 +496,16 @@ public boolean execute(int edge, List<GPXExtension> list) {
         return res;
     }
 
-    private boolean isVirtualNode(int node) {
+	private void addWayGeometry(int i, int lastIndex, PointList wayGeometryToFirstTowerNode, PointList wayGeometryToLastGPXPoint, EdgeMatch edgeMatch) {
```

Comment:
Any Idea to make that more clever than that? I don't really like it, but as we have the continue in the calling code, we need that piece of code two times...